Automattic/sensei
# Add Lesson Properties block

### Changes proposed in this Pull Request

Adds a _Lesson Properties_ block for lessons that displays lesson length and difficulty.

- The block saves attributes to post meta so won't have any content in the editor.
- I encountered the following error trying to save the block. It appears to be the same as [this Gutenberg issue](https://github.com/WordPress/gutenberg/issues/17864). [The fix](https://developer.wordpress.org/rest-api/extending-the-rest-api/modifying-responses/#invalid-stored-values) was to save `0` instead of `null` for `_quiz_has_questions`. I think this was more of a nice-to-have to keep the database cleaner, but let me know if I got that wrong:

![Screen Shot 2021-04-22 at 7 32 15 AM](https://user-images.githubusercontent.com/1190420/115707739-29875680-a33d-11eb-9abf-73661468d141.jpg)
- I encountered the following error when saving on Divi. The fix was to also register the `_lesson_course` post meta:

![divi error](https://user-images.githubusercontent.com/1190420/115707222-820a2400-a33c-11eb-8895-5a95e06b39d6.jpg)
- Fixed NaN displaying when the number input control was cleared. It's now set to its minimum value, if applicable, or 0. (Also applicable to Grading and Number of Questions fields for quizzes).
- The _Lesson Information_ metabox is not displayed when this block is present.
- The block is added to the lesson template by default.

### Testing instructions

* Create a new lesson and add the Lesson Properties block.
* Try setting different values for Length and Difficulty.
* When length is 0 the length should not be displayed on the frontend (it's still displayed in the editor).
* When difficulty is _None_ the difficulty should not be displayed on the frontend (it's still displayed in the editor).
* Ensure the _Lesson Information_ metabox is not shown.
* Remove the block and ensure the _Lesson Information_ metabox is shown.

### Screenshots
![Screen Shot 2021-04-30 at 9 59 57 AM](https://user-images.githubusercontent.com/1190420/116705726-d9db0780-a99a-11eb-9ecc-c269cd14e2ad.jpg)

### To Do
- [x] Block icon
